napari/_vispy/vispy_surface_layer.py

```diff
@@ -17,12 +18,6 @@ def __init__(self, layer):
 
         super().__init__(layer, node)
```

Comment:
You still need to add this initialization

https://github.com/napari/napari/blob/master/napari/_vispy/vispy_image_layer.py#L25-L28

Reply:
somehow I missed these attribute changes/additions to the Image Layer in that previous PR.  Tell me again why we need to duplicate what seem to be model properties on the vispy classes?  Is this because the event.value itself doesn't contain all of the necessary info?

Reply:
in some ways yes, but we'll be able to drop them again after the gamma PRs go through at vispy https://github.com/vispy/vispy/pull/1842 and https://github.com/vispy/vispy/pull/1844.

The issue now is that the vispy node only tracks the actual `cmap` which is a combination of the `_raw_cmap` and the `_gamma` value according to the CPU computation from before. When either the colormap or the gamma alone changes you need to access the value of the other variable now to get the correct final colormap. Once all this is within vispy that won't be necessary anymore and we'll just be able to update `node.gamma` and `node.cmap` and everything will work, I think!!

Does that checkout? It will be a good thing to verify before the next vispy release as you finalize that last PR

Reply:
that checkouts for gamma and cmap, but what about iso_threshold? and `_attenuation`?  I guess in general I missed the fact that changing the event handler required us to duplicate these model values at all.  in other words, it shouldn't require a PR to an external library to avoid that pattern, right?  like, maybe for complex cases like this, the event value itself needs to get more complex?

will open a different issue for this, but a related thing I'm sort of mourning is the loss of the `event.source` attribute after the event handler passes it on (I often used that for my own connections).  I can certainly just make my own connections without going through the event_handler ... we can discuss elsewhere.

Reply:
For `_attenuation` and `_iso_threshold` I actually think it was my bad design when I made my PR to vispy without adding a separate `attenuation` parameter, and instead overloading their `threshold` parameter that got us here. I might want to try and fix that before their next release, but if we don't fix in vispy, I think maintaining some state on the view like this is still ok. Much better to just maintain these two floats for a complex case than maintain the whole layer model, right?

As to the `event.source` stuff, we can discuss at team meeting. There's also some interesting stuff coming to light during in #1391 as I add the EVH to the layerlist that will be worth discussing too.